### PR TITLE
Add a callback that uses python logging

### PR DESCRIPTION
**Pull request by alikins**
##### SUMMARY

Add a callback that uses python logging

This also attempts to track playbook/play/task context so it can be provided to log handlers and formatters.

##### ISSUE TYPE

 - Feature Pull Request
 

##### COMPONENT NAME

lib/ansible/plugins/callback/stdlog.py
##### ANSIBLE VERSION

```
ansible 2.4.0 (stdlog_callback_new fbe2123aae) last updated 2017/06/15 15:09:38 (GMT -400)
  config file = None
  configured module search path = [u'/home/adrian/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/adrian/src/ansible/lib/ansible
  executable location = /home/adrian/src/ansible/bin/ansible
  python version = 2.7.13 (default, May 10 2017, 20:04:28) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by alikins** [comment]
test comment

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/28469/40/console

The relevant error appears to be:

```
2017-07-13 04:27:25 Missing MAGICKEYWORD in output.
```

File: lib/ansible/plugins/callback/stdlog.py [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+from ansible.plugins.callback import CallbackBase
+
+# import logging_tree
+
+
+# NOTE: in Ansible 1.2 or later general logging is available without
+# this plugin, just set ANSIBLE_LOG_PATH as an environment variable
+# or log_path in the DEFAULTS section of your ansible configuration
+# file.  This callback is an example of per hosts logging for those
+# that want it.
```
Comment (abadger) [inline lib/ansible/plugins/callback/stdlog.py:38]: is this comment right?

File: lib/ansible/plugins/callback/stdlog.py [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+import logging
+import logging.handlers
+import os
+import pprint
+import re
+
+# from ansible.utils.unicode import to_bytes
+from ansible.plugins.callback import CallbackBase
+
+# import logging_tree
```
Comment (abadger) [inline lib/ansible/plugins/callback/stdlog.py:31]: Go ahead and remove the commented out imports

File: lib/ansible/plugins/callback/stdlog.py [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+
+
+# NOTE: in Ansible 1.2 or later general logging is available without
+# this plugin, just set ANSIBLE_LOG_PATH as an environment variable
+# or log_path in the DEFAULTS section of your ansible configuration
+# file.  This callback is an example of per hosts logging for those
+# that want it.
+
+DEBUG_LOG_FORMAT = "%(asctime)s [%(name)s %(levelname)s %(hostname)s %(playbook)s] pid=%(process)d %(funcName)s:%(lineno)d - %(message)s"
```
Comment (bcoca) [inline lib/ansible/plugins/callback/stdlog.py:40]: I would make these configurable options, by adding them as options to DOCUMENTATION entry you can then use `self._plugin_options['optionname']` to access them

Comment (bcoca) [reply lib/ansible/plugins/callback/stdlog.py:40]: also .. add DOCUMENTATION entry

File: lib/ansible/plugins/callback/stdlog.py [hunk 4]
```diff
@@ -0,0 +1,395 @@
+# (C) 2012, Michael DeHaan, <michael.dehaan@gmail.com>
```
Comment (bcoca) [inline lib/ansible/plugins/callback/stdlog.py:1]: use new short Copyright/license

**Comment by bcoca** [comment]
I woudl also make this take over for the current 'ansbile_log'

**Comment by bcoca** [comment]
Hi!

Thank you very much for your submission to Ansible. It sincerely means a lot to us that you've taken the time to contribute.

Per recent announcements at AnsibleFest Atlanta 2019, going forward we are no longer accepting new one off modules/plugins, or modules/plugins that create a new category.

However, we recommend looking into providing this functionality through Ansible Galaxy via Ansible Collections. You can find more information about collections at https://docs.ansible.co
m/ansible/devel/dev_guide/developing_collections.html.

Thank you once again for this and your interest in Ansible!

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.freenode.net
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Pull request closed without merging**